### PR TITLE
feat: sort keys before calculating checksum

### PR DESCRIPTION
**Pull request by KSXGitHub**
fixes https://github.com/pnpm/pnpm/issues/6824

### CODE REVIEW

File: pkg-manager/core/src/install/index.ts [hunk 1]
```diff
@@ -749,8 +749,17 @@ function getOutdatedLockfileSetting (
...
+  const entries = Object
+    .entries(obj)
+    .sort(([k1, _v1], [k2, _v2]) => k1.localeCompare(k2, 'en'))
+    .map(([k, v]) => [k, sortObjectKeys(v)])
+  return Object.fromEntries(entries) as Obj
+}
+
 export function createObjectChecksum (obj: unknown) {
-  const s = JSON.stringify(obj)
+  const s = JSON.stringify(sortObjectKeys(obj))
```
Comment (zkochan) [inline pkg-manager/core/src/install/index.ts]: use the sort-keys library. It is already used in other places.

File: .changeset/quiet-brooms-serve.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@pnpm/core": patch
+---
```
Comment (zkochan) [inline .changeset/quiet-brooms-serve.md:4]: always include the pnpm project too in the list of changed projects.

File: pkg-manager/core/src/install/index.ts (hunk 1 above)
Comment (KSXGitHub) [reply pkg-manager/core/src/install/index.ts]: How this fix this error when test?

```text
SyntaxError: Cannot use import statement outside a module
```

**Review by zkochan — APPROVED** [review]

**Comment by gerbyzation** [comment]
Maybe this should have been marked as a breaking change? Ran into a situation where our pipelines are breaking as we use pnpm@8.6.10 w/ this fix included in some places, but not all. This causes different checksums to be generated:
```
ERR_PNPM_LOCKFILE_CONFIG_MISMATCH  Cannot proceed with the frozen installation. The current "packageExtensionsChecksum" configuration doesn't match the value found in the lockfile
```
Consistently using `>=8.6.10` resolves this issue.

**Pull request merged**